discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -12,15 +12,28 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+use futures::future::join_all;
+use itertools::Itertools;
+use std::time::Duration;
 use warp::{hyper::StatusCode, reject::Reject, Rejection};
 
-use xayn_discovery_engine_ai::{utils::rank, GenericError};
+use xayn_discovery_engine_ai::{
+    compute_coi_relevances,
+    system_time_now,
+    utils::rank,
+    CoiContextError,
+    GenericError,
+    PositiveCoi,
+};
 
 use crate::{
     db::Db,
-    models::{InteractionRequestBody, UserId},
+    models::{Error, InteractionRequestBody, UserId},
 };
 
+const MAX_COIS: usize = 5;
+const MAX_DOCUMENTS_COUNT: usize = 100;
```

Comment:
❓ should we make them associated constants of the `AppState` or even config values of the `AppState`?